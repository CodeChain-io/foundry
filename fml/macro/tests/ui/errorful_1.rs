extern crate fml_macro;

#[fml_macro::service]
pub trait MyHandle: Service {
    fn fn_1(&self, a: i32, b: String) -> String;
    fn fn_2(a: i32, b: Vec<u8>) -> Box<dyn MyHandle>; // no &self
}

fn main() {

}
