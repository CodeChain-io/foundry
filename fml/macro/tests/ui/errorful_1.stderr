error: All your method must take &self
 --> $DIR/errorful_1.rs:6:5
  |
6 |     fn fn_2(a: i32, b: Vec<u8>) -> Box<dyn MyHandle>; // no &self
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
